nilearn/plotting/html_connectome.py

```diff
@@ -27,49 +65,137 @@ def _prepare_line(edges, nodes):
+    """
+    if isinstance(marker_color, str) and marker_color == 'auto':
+        colors = mpl_cm.Set2(np.linspace(0, 1, number_of_nodes))
+    elif isinstance(marker_color, str):
+        cmap = mpl_colors.ListedColormap([marker_color] * number_of_nodes)
+        colors = cmap(np.arange(cmap.N))
+    else:
+        cmap = mpl_colors.ListedColormap(marker_color)
+        colors = cmap(np.arange(cmap.N))
+
+    colors = colors[:, :3]
+    colors = np.asarray(colors * 255, dtype='uint8')
+    colors = ['#{:02x}{:02x}{:02x}'.format(*row) for row in colors]
+
+    return colors
+
+
+def _prepare_lines_metadata(adjacency_matrix, coords, threshold, cmap, symmetric_cmap):
+    """
+    Generate metadata related to lines for _connectome_view plot
+
+    Parameters
+    ----------
+    adjacency_matrix : ndarray, shape=(n_nodes, n_nodes)
+        The weights of the edges.
+
+    coords : ndarray, shape=(n_nodes, 3)
+        The coordinates of the nodes in MNI space.
+
+    threshold : str, number or None, optional
+        If None, no thresholding.
+        If it is a number only connections of amplitude greater
+        than threshold will be shown.
+        If it is a string it must finish with a percent sign,
+        e.g. "25.3%", and only connections of amplitude above the
+        given percentile will be shown.
+
+    cmap : str or matplotlib colormap, optional
+        Colormap to use. Default=cm.bwr.
+
+    symmetric_cmap : bool, optional
+        Make colormap symmetric (ranging from -vmax to vmax).
+        Default=True.
+
+    Returns
+    -------
+    coordinates : dict
+        Dictionary containing base64 values for each axis
+    """
+    lines_metadata = {}
+
     adjacency_matrix = np.nan_to_num(adjacency_matrix, copy=True)
     colors = colorscale(
         cmap, adjacency_matrix.ravel(), threshold=threshold,
         symmetric_cmap=symmetric_cmap)
-    connectome['colorscale'] = colors['colors']
-    connectome['cmin'] = float(colors['vmin'])
-    connectome['cmax'] = float(colors['vmax'])
+    lines_metadata['line_colorscale'] = colors['colors']
+    lines_metadata['line_cmin'] = float(colors['vmin'])
+    lines_metadata['line_cmax'] = float(colors['vmax'])
     if threshold is not None:
         adjacency_matrix[
             np.abs(adjacency_matrix) <= colors['abs_threshold']] = 0
     s = sparse.coo_matrix(adjacency_matrix)
     nodes = np.asarray([s.row, s.col], dtype=int).T
     edges = np.arange(len(nodes))
     path_edges, path_nodes = _prepare_line(edges, nodes)
-    connectome["_con_w"] = encode(np.asarray(s.data, dtype='<f4')[path_edges])
-    c = coords[path_nodes]
+    lines_metadata["_con_w"] = encode(np.asarray(s.data, dtype='<f4')[path_edges])
+
+    line_coords = coords[path_nodes]
+
+    lines_metadata = {
+        **lines_metadata,
+        **_encode_lines_coordinates(line_coords)
+    }
+
+    return lines_metadata
+
+
+def _prepare_markers_metadata(coords, marker_size, marker_color):
```

Comment:
It seems this preparation of markers is duplicated:
- when plotting a full connectome it appears here (in `_prepare_markers_metadata`)
- when plotting markers only half of it is inlined in `view_markers` and the rest in `_get_markers`.

but the logic in both cases in the same and the only important difference is that "markers_only" needs to be set to `True` in the second case.

could we remove `_get_markers` and use `_prepare_markers_metadata` in both cases?